http-filter-example/http_filter.cc

```diff
@@ -23,55 +23,57 @@ HttpSampleDecoderFilterConfig::HttpSampleDecoderFilterConfig(
 
 HttpSampleDecoderFilter::HttpSampleDecoderFilter(HttpSampleDecoderFilterConfigSharedPtr config)
     : config_(config) {
-  // populate list of valid operations, implementation will be changed in the next few commits
-  operations_.insert("set-header");
   
   const std::string header_config = headerValue();
 
-  // find the position of the first space character
-  const size_t spacePos = header_config.find(' ');
-  if (spacePos == std::string::npos) {
-    fail("no arguments provided");
-    setError(1);
-    return;
-  }
+  // split by operation (comma delimited config)
+  auto operations = StringUtil::splitToken(header_config, "\n", false, true);
 
-  // can't be a string view
-  const auto operation = header_config.substr(0, spacePos);
+  // process each operation
+  for (auto const& operation : operations) {
+    auto tokens = StringUtil::splitToken(operation, " ");
+    if (tokens.size() < 3) {
+      fail("too few arguments provided");
+      setError(1); // TODO: set error based on globally defined macros
+      return;
+    }
 
-  // error checking -- invalid operation
-  if (operations_.find(operation) == operations_.end()) {
-    fail("invalid operation provided");
-    setError(1);
-    return;
-  }
+    // determine if it's request/response
+    if (tokens[0] != "http-request" && tokens[0] != "http-response") {
+      fail("first argument must be <http-response/http-request>");
+      setError(1); // TODO: set error based on globally defined macros
+      return;
+    }
+    bool isRequest = (tokens[0] == "http-request");
 
-  const std::string args = header_config.substr(spacePos + 1);
+    // TODO: determine the operation type (right now I'm assuming that it's always a set-header operation)
 
-  // create a string stream to iterate over the arguments
-  std::istringstream iss(args);
-  std::vector<std::string> values;
-  std::string value;
+    // make new header processor
+    SetHeaderProcessor* processor = new SetHeaderProcessor();
```

Comment:
This can be a `std::unique_ptr` or `std::shared_ptr` to avoid having to manually free it.
We should try to always use `unique_ptr` and `shared_ptr`, and never `new` or `malloc`, if possible. Memory leaks from manual memory management are a huge source of bugs in C++

Reply:
I _think_ `unique_ptr` works, since I think it only ever needs 1 owner. But I'm not 100% sure of that; if not, `shared_ptr` is fine